### PR TITLE
Improve time tracking overview

### LINKED ISSUE #7708: Remove Google Chart and Improve Time Tracking Overview
## Detailed Description
- Find better chart for time-line view
- Improve table in time tracking overview, see https://scm.slack.com/archives/C5AKLAV0B/p1708951683131989?thread_ts=1704359796.147499&cid=C5AKLAV0B

Follow-up for #5375

### PR DESCRIPTION
**Pull request by dieknolle3333**
### URL of deployed dev instance (used for testing):
- https://timetrackingexpandtable.webknossos.xyz/

### Steps to test:
- Make sure to have some tracked times in explorative annotations and tasks
- log in as admin-user and go to "Time tracking overview" through navbar
- query time tracking entries
- expand details for users
- test download buttons 
- test as non-priviledged user (should only see yourself)
- test without login (view shouldn't be accessible) 

### TODOs:
- [x] see issue
- [x] think about csv export structure
- [x] fix types
- [x] adjust tests to new param: getTimeTrackingForUserSpans, getTimeTrackingForUser
- [x] fix that non-admin user cannot see themselves because "page doesnt exist"
- [x] fix that expendable button is in wrong position after adjusting filters
- [x] make sure new project line looks good with segmentation stats

### Issues:
- fixes #7708 

------
(Please delete unneeded items, merge only when none are left open)
- [x] Updated [changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [x] Removed dev-only changes like prints and application.conf edits
- [x] Considered [common edge cases](../blob/master/.github/common_edge_cases.md)

### CODE REVIEW

File: frontend/javascripts/admin/statistic/time_tracking_detail_view.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import { useFetch } from "libs/react_helpers";
+import React from "react";
+import { AnnotationTypeFilterEnum } from "./project_and_annotation_type_dropdown";
+import { getTimeTrackingForUserSummedPerAnnotation } from "admin/admin_rest_api";
+import dayjs from "dayjs";
+import { Col, Divider, Row } from "antd";
+import { formatMilliseconds } from "libs/format_utils";
+import _ from "lodash";
+import { APITimeTrackingPerAnnotation } from "types/api_flow_types"; // TODO adjust type based on request
```
Comment (dieknolle3333) [inline frontend/javascripts/admin/statistic/time_tracking_detail_view.tsx:9]: remove when done

**Review by normanrz — APPROVED** [review]
Backend code looks good :+1:

File: CHANGELOG.unreleased.md [hunk 2]
```diff
@@ -24,6 +24,7 @@ For upgrade instructions, please check the [migration guide](MIGRATIONS.released
 - When proofreading segments and merging two segments, the segment item that doesn't exist anymore after the merge is automatically removed. [#7729](https://github.com/scalableminds/webknossos/pull/7729)
 - Changed some internal APIs to use spelling dataset instead of dataSet. This requires all connected datastores to be the latest version. [#7690](https://github.com/scalableminds/webknossos/pull/7690)
 - Toasts are shown until WEBKNOSSOS is running in the active browser tab again. Also, the content of most toasts that show errors or warnings is printed to the browser's console. [#7741](https://github.com/scalableminds/webknossos/pull/7741)
+- In the time tracking view, all annotations and tasks can be shown for each user by expanding the table. The individual time spans spent with a task or annotating an explorative annotation can be accessed via CSV export. The detail view including a chart for the individual spans was omitted. [#7733](https://github.com/scalableminds/webknossos/pull/7733)
```
Comment (philippotto) [inline CHANGELOG.unreleased.md:27]:
```suggestion
- In the time tracking view, all annotations and tasks can be shown for each user by expanding the table. The individual time spans spent with a task or annotating an explorative annotation can be accessed via CSV export. The detail view including a chart for the individual spans has been removed. [#7733](https://github.com/scalableminds/webknossos/pull/7733)
```

File: frontend/javascripts/admin/admin_rest_api.ts [hunk 3]
```diff
@@ -1967,7 +1987,7 @@ export async function getTimeEntries(
   teamIds: string[],
   selectedTypes: AnnotationTypeFilterEnum,
   projectIds: string[],
-) {
+): Promise<Array<APITimeTrackingPerUser>> {
```
Comment (philippotto) [inline frontend/javascripts/admin/admin_rest_api.ts:1990]: :+1:

File: frontend/javascripts/types/api_flow_types.ts [hunk 4]
```diff
@@ -537,15 +537,36 @@ export type APIAnnotationWithTask = APIAnnotationBase & {
...
+  projectName: string | undefined;
+  timeMillis: number;
+  annotationLayerStats: Array<TracingStats>;
+};
+export type APITimeTrackingPerUser = {
+  user: {
+    id: string;
+    firstName: string;
+    lastName: string;
+    email: string;
```
Comment (philippotto) [inline frontend/javascripts/types/api_flow_types.ts:552]: can you define this type as an extension of `APIUserCompact` (similar to `APIUserBase`) in api_flow_types.ts?

File: frontend/javascripts/oxalis/model/accessors/annotation_accessor.ts [hunk 5]
```diff
@@ -97,23 +98,30 @@ export function getCombinedStats(tracing: Tracing): CombinedTracingStats {
...
-    const { stats } = annotationLayer;
+export function aggregateStatsForAllLayers(
+  stats: Array<TracingStats | EmptyObject>,
+): CombinedTracingStats {
+  const aggregatedStats: TracingStatsHelper = {};
 
-    if ("treeCount" in stats) {
-      const { treeCount, nodeCount, edgeCount, branchPointCount } = stats;
+  for (const annotationLayerStats of stats) {
+    if (Object.keys(annotationLayerStats).includes("treeCount")) {
```
Comment (philippotto) [inline frontend/javascripts/oxalis/model/accessors/annotation_accessor.ts:112]: why this change? :thinking: is `in` not okay?

File: frontend/javascripts/admin/statistic/time_tracking_detail_view.tsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+      annotationRows = annotationRows.concat(tableRows);
+    } else {
+      // tasks
+      taskRows.push(
+        <Row style={{ fontWeight: "bold", margin: 20 }}>
+          <Col>{project}</Col>
+        </Row>,
+      );
+      const tableRows = loggedTimes.map((timeEntry) => (
+        <Row style={{ lineHeight: 3.3 }}>
```
Comment (philippotto) [inline frontend/javascripts/admin/statistic/time_tracking_detail_view.tsx:59]: why does this row have a special lineHeight?

**Review by philippotto — COMMENTED** [review]
nice stuff :+1:

during testing I noticed that the user-specific CSV contains timestamps, such as "Fri, 19 Apr 2024 15:42:55 GMT". wouldn't unix timestamps make more sense as they are easier to parse? not sure whether they are coming from back-end or front-end...

File: frontend/javascripts/oxalis/model/accessors/annotation_accessor.ts (hunk 5 above)
Comment (dieknolle3333) [reply frontend/javascripts/oxalis/model/accessors/annotation_accessor.ts:112]: oh, this is a leftover from when somewhere in the process it wasnt working :)

File: frontend/javascripts/admin/statistic/time_tracking_detail_view.tsx (hunk 6 above)
Comment (dieknolle3333) [reply frontend/javascripts/admin/statistic/time_tracking_detail_view.tsx:59]: because there are fewer segments stats, the lines would be much lower, and I thought it looked weird (in the picture I removed the styling). Will add a comment :)

Comment (dieknolle3333) [reply frontend/javascripts/admin/statistic/time_tracking_detail_view.tsx:59]: image

**Comment by dieknolle3333** [comment]
>  wouldn't unix timestamps make more sense as they are easier to parse? not sure whether they are coming from back-end or front-end...

I had added them in the frontend in the first place, but changed them to unix timestamp now :)

File: frontend/javascripts/admin/statistic/time_tracking_detail_view.tsx (hunk 6 above)
Comment (dieknolle3333) [reply frontend/javascripts/admin/statistic/time_tracking_detail_view.tsx:59]: wait, I am not sure this isn't due to the way I am testing

Comment (dieknolle3333) [reply frontend/javascripts/admin/statistic/time_tracking_detail_view.tsx:59]: ah, I think the segment count is never shown for tasks because its not possible to create segments in tasks, right?

Comment (fm3) [reply frontend/javascripts/admin/statistic/time_tracking_detail_view.tsx:59]: depending on the settings in the task type selected during task creation, it should be possible to create segments in tasks. But it may well be that the sampleTaskType in the development setup is set to skeleton-only.
image

Comment (dieknolle3333) [reply frontend/javascripts/admin/statistic/time_tracking_detail_view.tsx:59]: alright, I will test the styling with skeleton and volume then :)

Comment (dieknolle3333) [reply frontend/javascripts/admin/statistic/time_tracking_detail_view.tsx:59]: yeah, I will need to update this 
image

Comment (dieknolle3333) [reply frontend/javascripts/admin/statistic/time_tracking_detail_view.tsx]: I adjusted the styling so that it seems to look okay in both cases :)

File: package.json [hunk 7]
```diff
@@ -193,7 +193,6 @@
     "react-dom": "^16.9.0",
     "react-dropzone": "^11.3.1",
     "react-flow-renderer": "^10.3.16",
-    "react-google-charts": "^2.0.0",
```
Comment (dieknolle3333) [inline package.json:196]: i simply ran `yarn remove react-google-charts`, I hope its good the way it is now :)

**Comment by dieknolle3333** [comment]
no worries, thanks for looking into it! I will merge it on monday then :)
